juju/juju
# Default model config is now per cloud

## Description of change

Default model config used to be stored globally for all models in the controller, since all models shared the same cloud. Now that controllers support models in different clouds, the default config needs to be per cloud.

There's 4 commits:
- state changes
- facade changes
- CLI changes
- upgrade steps

## QA steps

bootstrap 2.5.1 with default cloud config
upgrade to this branch
check that the data in globalSettings collection has been upgraded
run juju model-defaults in various scenarios, checking that combinations of no args, args with cloud, args with cloud/region etc work
